### PR TITLE
fix: fix sign-attestations output path

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Fixes https://github.com/slsa-framework/slsa-github-generator/issues/1472

Working on adding `main` tests that mock `sigstore` for this one. For https://github.com/slsa-framework/slsa-github-generator/issues/1471

### CODE REVIEW

File: .github/actions/sign-attestations/src/main.ts [hunk 1]
```diff
@@ -50,11 +56,10 @@ async function run(): Promise<void> {
         );
         const bundleStr = JSON.stringify(bundle);
         // We detect path traversal for safeOutputFolder, so this should be safe.
-        const outputPath = path.join(
+        const outputPath = `${path.join(
```
Comment (laurentsimon) [inline .github/actions/sign-attestations/src/main.ts:59]: I dont think you need this. The following should work, no:
```
const outputPath = path.join(
           safeOutputFolder
           path.basename(file) + ".sigstore",
        );
```

**Review by laurentsimon — APPROVED** [review]
Thanks!

File: .github/actions/sign-attestations/src/main.ts (hunk 1 above)
Comment (asraa) [reply .github/actions/sign-attestations/src/main.ts:59]: npm style doesn't like string concatenation and prefers templating D:

/home/asraa/git/slsa-github-generator/.github/actions/sign-attestations/src/main.ts
  61:11  error  Unexpected string concatenation  prefer-template

But it's cleaner to do similar to what you suggested:
```
        const outputPath = path.join(
          safeOutputFolder,
          `${path.basename(fpath)}.sigstore`);
```

File: .github/actions/sign-attestations/dist/index.js [hunk 2]
```diff
@@ -37127,7 +37133,7 @@ function run() {
                     const bundle = yield sigstore.sigstore.signAttestation(buffer, payloadType, signOptions);
                     const bundleStr = JSON.stringify(bundle);
                     // We detect path traversal for safeOutputFolder, so this should be safe.
-                    const outputPath = path.join(safeOutputFolder, path.basename(file), ".sigstore");
+                    const outputPath = `${path.join(safeOutputFolder, path.basename(fpath))}.sigstore`;
```
[the remark this answers is not in the document]
Comment (asraa) [reply .github/actions/sign-attestations/dist/index.js:37136]: this is the generated

File: .github/actions/sign-attestations/src/main.ts (hunk 1 above)
Comment (laurentsimon) [reply .github/actions/sign-attestations/src/main.ts:59]: Got it, thanks!

**Review by asraa — COMMENTED** [review]
updated, also realized I checked in a spurious outputs/ directory created in local test so deleted here.

**Pull request merged**